server/channels/app/export.go

```diff
@@ -829,7 +829,12 @@ func (a *App) exportAllDirectChannels(ctx request.CTX, job *model.Job, writer io
 				return err
 			}
 
-			channelLine := ImportLineFromDirectChannel(channel, favoritedBy)
+			shownBy, err := a.buildShownByList(channel)
+			if err != nil {
+				return err
+			}
+
+			channelLine := ImportLineFromDirectChannel(channel, favoritedBy, shownBy)
```

Comment:
Hmm, I'm not sure exporting a deleted channel would cause issue but we can add a test case for that. But if the members are deleted (hard deletion) without deleting the app level foreign keys (unfortunately), yes there will be issues.

Reply:
Gave it a look and found two issues:
- https://mattermost.atlassian.net/browse/MM-60082 -> Deleting a single user prevents a GM being exported
- https://mattermost.atlassian.net/browse/MM-60083 -> Deleting a single user from a DM should prevent that channel to be exported.

Reply:
Thanks. Do we know if any of the above could be affecting the current customer migration?

Reply:
Not that I'm aware of, it's a case if the customer has permanently deleted a user. But haven't heard anything from them so far.